src/ansiblelint/rules/__init__.py

```diff
@@ -114,7 +114,7 @@ def matchtasks(self, file: Lintable) -> List[MatchError]:
 
             if 'action' not in task:
                 continue
-            result = self.matchtask(task)
+            result = self.matchtask(file, task)
```

Comment:
Clearly these calls use incorrect order or arguments. I hope they fail the linting as I do not want to see others doing the same mistake.